projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/order-amend.service.ts

```diff
@@ -0,0 +1,128 @@
+import { formatCurrency, getCurrencySymbol } from '@angular/common';
+import { Injectable } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Order, OrderEntry, Price } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map, switchMap, tap } from 'rxjs/operators';
+import { OrderDetailsService } from '../order-details/order-details.service';
+import { OrderAmendType } from './order-amend.model';
+
+function ValidateQuantity(control: FormControl) {
+  let q = 0;
+  Object.keys(control.value).forEach(key => (q += control.value[key]));
+
+  return q > 0 ? null : { required: true };
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrderAmendService {
+  protected amendType: OrderAmendType;
+  protected form: FormGroup;
+  protected lang: 'string';
+
+  constructor(protected orderDetailsService: OrderDetailsService) {}
+
+  /**
+   * Returns entries for the given order.
+   */
+  getEntries(): Observable<OrderEntry[]> {
```

Comment:
Couldn't you have left this abstract if you made the class abstract?

Reply:
yes, moved it back to abstract. I was abstract, but I was fighting with DI and hence made the "temp" change.